libraries/opensk/src/env/test/crypto.rs

```diff
@@ -12,208 +12,12 @@
-    }
-
-    fn to_der(&self) -> Vec<u8> {
-        self.signature.to_asn1_der()
-    }
-}
-
-#[cfg(test)]
-mod test {
-    use super::*;
-    use crate::api::crypto::ecdh::{
-        PublicKey as EcdhPublicKey, SecretKey as EcdhSecretKey, SharedSecret,
-    };
-    use crate::api::crypto::ecdsa::{PublicKey as EcdsaPublicKey, SecretKey as EcdsaSecretKey};
-    use crate::env::test::TestEnv;
-
-    #[test]
-    fn test_shared_secret_symmetric() {
-        let mut env = TestEnv::default();
-        let private1 = TestEcdhSecretKey::random(env.rng());
-        let private2 = TestEcdhSecretKey::random(env.rng());
-        let pub1 = private1.public_key();
-        let pub2 = private2.public_key();
-        let shared1 = private1.diffie_hellman(&pub2);
-        let shared2 = private2.diffie_hellman(&pub1);
-        assert_eq!(shared1.raw_secret_bytes(), shared2.raw_secret_bytes());
-    }
-
-    #[test]
-    fn test_ecdh_public_key_from_to_bytes() {
-        let mut env = TestEnv::default();
-        let first_key = TestEcdhSecretKey::random(env.rng());
-        let first_public = first_key.public_key();
-        let mut x = [0; EC_FIELD_BYTE_SIZE];
-        let mut y = [0; EC_FIELD_BYTE_SIZE];
-        first_public.to_coordinates(&mut x, &mut y);
-        let new_public = TestEcdhPublicKey::from_coordinates(&x, &y).unwrap();
-        let mut new_x = [0; EC_FIELD_BYTE_SIZE];
-        let mut new_y = [0; EC_FIELD_BYTE_SIZE];
-        new_public.to_coordinates(&mut new_x, &mut new_y);
-        assert_eq!(x, new_x);
-        assert_eq!(y, new_y);
-    }
-
-    #[test]
-    fn test_sign_verify() {
-        let mut env = TestEnv::default();
-        let private_key = TestEcdsaSecretKey::random(env.rng());
-        let public_key = private_key.public_key();
-        let message = [0x12, 0x34, 0x56, 0x78];
-        let signature = private_key.sign(&message);
-        assert!(public_key.verify(&message, &signature));
-    }
-
-    #[test]
-    fn test_ecdsa_secret_key_from_to_bytes() {
-        let mut env = TestEnv::default();
-        let first_key = TestEcdsaSecretKey::random(env.rng());
-        let mut key_bytes = [0; EC_FIELD_BYTE_SIZE];
-        first_key.to_slice(&mut key_bytes);
-        let second_key = TestEcdsaSecretKey::from_slice(&key_bytes).unwrap();
-        let mut new_bytes = [0; EC_FIELD_BYTE_SIZE];
-        second_key.to_slice(&mut new_bytes);
-        assert_eq!(key_bytes, new_bytes);
-    }
-}
+use crate::api::crypto::software_crypto;
+
+pub use software_crypto::{
+    SoftwareCrypto as TestCrypto, SoftwareEcdh as TestEcdh,
+    SoftwareEcdhPublicKey as TestEcdhPublicKey, SoftwareEcdhSecretKey as TestEcdhSecretKey,
+    SoftwareEcdhSharedSecret as TestEcdhSharedSecret, SoftwareEcdsa as TestEcdsa,
+    SoftwareEcdsaPublicKey as TestEcdsaPublicKey, SoftwareEcdsaSecretKey as TestEcdsaSecretKey,
+    SoftwareEcdsaSignature as TestEcdsaSignature,
+};
```

Comment:
This is to avoid renaming those usage?

Reply:
I used the shorthands `EcdsaSk` etc in all places where used, so the implementations don't have to be accessed outside of `Env` anymore. This makes the whole file obsolete.